### PR TITLE
feat(AM-3563):  added validations

### PR DESCRIPTION
**Pull request by priyank-upadhyay**

### CODE REVIEW

File: service/slice_config_webhook_validation.go [hunk 1]
```diff
@@ -247,7 +247,16 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig) *fie
...
-func validateQosProfile(sliceConfig *controllerv1alpha1.SliceConfig) *field.Error {
+func validateQosProfile(ctx context.Context, sliceConfig *controllerv1alpha1.SliceConfig) *field.Error {
+	if len(sliceConfig.Spec.StandardQosProfileName) > 0 && len(sliceConfig.Spec.QosProfileDetails.QueueType) > 0 {
+		return field.Invalid(field.NewPath("Spec").Child("StandardQosProfileName"), sliceConfig.Spec.StandardQosProfileName, "StandardQosProfileName cannot be set when QosProfileDetails is set")
+	}
+	if len(sliceConfig.Spec.StandardQosProfileName) <= 0 && len(sliceConfig.Spec.QosProfileDetails.QueueType) <= 0 {
+		return field.Invalid(field.NewPath("Spec").Child("StandardQosProfileName"), sliceConfig.Spec.StandardQosProfileName, "Either StandardQosProfileName or QosProfileDetails is required")
+	}
+	if len(sliceConfig.Spec.StandardQosProfileName) > 0 {
+		existsQosConfigFromStandardQosProfileName(ctx, sliceConfig.Namespace, sliceConfig.Spec.StandardQosProfileName)
```
Comment (PuranAdhikari) [inline service/slice_config_webhook_validation.go:258]: We need to throw an error if not found, right?

File: service/slice_config_webhook_validation.go [hunk 2]
```diff
@@ -247,7 +247,16 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig) *fie
 }
 
 // validateQosProfile is a function to validate the Qos(quality of service)profile of slice
-func validateQosProfile(sliceConfig *controllerv1alpha1.SliceConfig) *field.Error {
+func validateQosProfile(ctx context.Context, sliceConfig *controllerv1alpha1.SliceConfig) *field.Error {
+	if len(sliceConfig.Spec.StandardQosProfileName) > 0 && len(sliceConfig.Spec.QosProfileDetails.QueueType) > 0 {
```
Comment (PuranAdhikari) [inline service/slice_config_webhook_validation.go:251]: Are we doing `len()` function for string `StandardQosProfileName`? We can check the blank string "", right?

File: service/slice_config_webhook_validation.go (hunk 1 above)
Comment (priyank-upadhyay) [reply service/slice_config_webhook_validation.go:258]: oh. missed that.

File: service/slice_config_webhook_validation.go [hunk 3]
```diff
@@ -247,7 +247,20 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig) *fie
...
+		return field.Invalid(field.NewPath("Spec").Child("StandardQosProfileName"), sliceConfig.Spec.StandardQosProfileName, "StandardQosProfileName cannot be set when QosProfileDetails is set")
+	}
+	if len(sliceConfig.Spec.StandardQosProfileName) <= 0 && len(sliceConfig.Spec.QosProfileDetails.QueueType) <= 0 {
+		return field.Invalid(field.NewPath("Spec").Child("StandardQosProfileName"), sliceConfig.Spec.StandardQosProfileName, "Either StandardQosProfileName or QosProfileDetails is required")
+	}
+	if len(sliceConfig.Spec.StandardQosProfileName) > 0 {
+		exists := existsQosConfigFromStandardQosProfileName(ctx, sliceConfig.Namespace, sliceConfig.Spec.StandardQosProfileName)
+		if !exists {
+			return field.Invalid(field.NewPath("Spec").Child("StandardQosProfileName"), sliceConfig.Spec.StandardQosProfileName, "StandardQosProfileName not found.")
```
Comment (PuranAdhikari) [inline service/slice_config_webhook_validation.go:260]: Can we say, "SliceQoSConfig not found" instead of "StandardQosProfileName not found."?

**Review by PuranAdhikari — APPROVED** [review]

**Review by narmidm — APPROVED** [review]

**Pull request merged**